### ISSUE TITLE
Random world name will always be "Tom" in Chinese version

### ISSUE THREAD

**Issue by cainiaowu** [CleverRaven/Cataclysm-DDA#16129]
Tested with zh-cn/zh-tw/ja, press \* will generate random name, and it will only be "tom" (and related translated word).

Random character name will also generate "Tom" a lot ("Tom" only, without last name),  some where around 1 in 5.

English version does not have this issue.

0.c-12763-g100d5bd ,windows

**Comment by player0981234** [comment]
confirmed

**Comment by Night-Pryanik** [comment]
As for "Tom" world names, you should have at least one string with text like this:
`{"usage": "world", "name": "foo"},`
in `\data\names\zh_CN(TW).json`
See `\data\names\en.json` for reference.

**Comment by player0981234** [comment]
I searched and cannot find anything like that in the zh_CN and zh_TW files
this is probably the reason

**Comment by Night-Pryanik** [comment]
@player0981234 so you need to add the names manually.

**Comment by DLaboratory** [comment]
First of all, you mistakenly spelled random.
It's because there is no any world name available in `/data/names/zh_CN.json`, so the code will choose translated `Tom` as the world name in `name.cpp` line 111.
```C++
    // BUG, no matching name found.
    return std::string( _( "Tom" ) );
```

**Comment by DLaboratory** [comment]
Do you want to use translated English world names, like 亚伯拉罕, 利物浦, 阿贝尔, or to use native Chinese world names? I'll work on it if I have a clear idea.

**Comment by cainiaowu** [comment]
@DLaboratory I think we should unify the name generation to make those name translateable and include them in .mo, so it would be easier to do the vesion control on transifex side.

**Comment by kevingranade** [comment]
The reason we're doing it the way we are is that we don't want translated
names, we want completely seperate lists of names in each language.
Is there any particular reason to keep a simple list of names on transifex?

**Comment by cainiaowu** [comment]
I think add a note to tell the translator that these are names of people/world/city would be enough.

Keep the list on transifex so that translators dont need to create seperate pull requests for these name list when updating translation.

**Comment by kevingranade** [comment]
> I think adding a note to tell the translator that these are names of people/world/city would be enough.

Where would you want this note?
> Keep the list on transifex so that translators dont need to create seperate pull requests for these name lists when updating translation.

It's a list of names, it doesn't need to be kept in sync with anything. The reason the MOTD and credits files are separate is they have formatting requirements that gettext isn't going to enforce, if you can make it do so we can migrate them.

**Comment by cainiaowu** [comment]
> Where would you want this note?

Sometimes when translating vague string on transifex, I could see a section called "Developer note" with tags so that I know how to translate them properly.
![a](https://cloud.githubusercontent.com/assets/4143592/14696582/da835054-07ae-11e6-890a-487c3689d40a.jpg)
It would be really nice to have the python extractor extract the "usage" from a single names.json and write them as these comments/notes.
I would really like to help out but I am not a coder myself.

**Comment by kevingranade** [comment]
There isn't a string to attach the gettext style translator's note to
(except "tom" I guess?).
The problem here is this isn't translation, it's a list of names that need
to be provided for each language. As far as I know transifex doesn't have
anywhere to put such a thing.

**Comment by cainiaowu** [comment]
I don't know if I understand the problem correctly here.
But it it possible to treat name like those items in various .json files? 
With their "id" as untranslated English name and "name" as translatable field. 
If the translation is not provided, then it can fallback to its untranslated form. 
"usage" can be a comment section to provide context to translator.

**Comment by kevingranade** [comment]
If you did that, you'd have to have exactly as many names of every
language, why? Not to mention, most names don't even have a translation.

**Comment by BrettDong** [comment]
You should edit your description. Russian version has its own world names. Not all "non-English" locales are having this issue.

**Comment by kevingranade** [comment]
Fixed by #17614, for one of the chinese dialects at least.

**Comment by BrettDong** [comment]
Would you like to check your inbox?
